xbmc/epg/Epg.cpp

```diff
@@ -52,9 +50,7 @@ CEpg::CEpg(int iEpgID, const CStdString &strName /* = "" */, const CStdString &s
   m_pvrChannel = empty;
 }
 
-CEpg::CEpg(CPVRChannelPtr channel, bool bLoadedFromDb /* = false */) :
-    m_bChanged(!bLoadedFromDb),
-    m_bTagsChanged(false),
+CEpg::CEpg(CPVRChannelPtr channel) :
```

Comment:
Using const ref you violate the concept. If you want avoid unnecessary copy you have to use shared_ptr::swap in CEpg ctor, as below :
```
CEpg::CEpg(CPVRChannelPtr channel) :
m_bLoaded(false),
m_bUpdatePending(false),
m_iEpgID(channel->EpgID()),
m_strName(channel->ChannelName()),
m_strScraperName(channel->EPGScraper()),
m_pvrChannel(),
m_bUpdateLastScanTime(false)
{
  m_pvrChannel.swap(channel);
}
```